olive/model/handler/onnx.py

```diff
@@ -51,19 +60,33 @@ def __init__(
         self.inference_settings = inference_settings
         self.use_ort_extensions = use_ort_extensions
         self.onnx_file_name = onnx_file_name
+        self.external_initializers_file_name = external_initializers_file_name
+        self.constant_inputs_file_name = constant_inputs_file_name
 
         self.io_config = None
         self.graph = None
         self.all_graphs: Optional[List[GraphProto]] = None
 
-        # check for onnx file name since it will do validation
-        _ = self.model_path
+        # check for file names since it will automatically validate the paths
+        to_check = ["model_path", "external_initializers_path", "constant_inputs_path"]
+        for attr in to_check:
+            getattr(self, attr)
```

Comment:
>getattr(self, attr) [](http://example.com/codeflow?start=12&length=19)

when the error will be raised?

Reply:
Any of the errors from https://github.com/microsoft/Olive/blob/25a0b78949fe295fa0f7faabd30dc328c229f853/olive/model/utils/onnx_utils.py#L30 or https://github.com/microsoft/Olive/blob/25a0b78949fe295fa0f7faabd30dc328c229f853/olive/model/utils/onnx_utils.py#L66

This is to make sure the values are provided correctly. if external data, initializers or constant inputs, they should be files inside `model_path` which should be folder.